tests/unittests/tests-kernel_defines/tests-kernel_defines.c

```diff
@@ -77,12 +77,24 @@ static void test_declare_constant(void)
     static_assert(sizeof(test_array) == 3, "test_array should be 3 bytes long");
 }
 
+static void test_is_compile_time_constant(void)
+{
+    /* These test might fail on non-GCC-non-clang compilers. We don't support
+     * any of those (yet), but this test might need adaption in the future */
+    int actual_constant = (int)(42 * 1337 / 3.14159);
```

Comment:
```suggestion
    unsigned actual_constant = (unsigned)(42 * 1337 / 3.14159);
```

Reply:
That was not enough to make GCC 11.2 for the AVR happy, I changed the computation a bit to not overflow. In addition, I moved this into at `static inline` function to make it a bit less trivial, so that we have a decent base-line on what we expect from compilers to detect as suitable for constant-folding.